### ISSUE TITLE
Adding a note to the favorite list does not work in grid view

### ISSUE THREAD

**Issue by chrgernoe** [nextcloud/notes-android#2529]
### This issue respects the following points:

- [x] I have read the [FAQ](https://github.com/nextcloud/notes-android/blob/master/FAQ.md)
- [x] I have searched for existing issues
- [x] The issue affects the Notes Android app, neither the [Notes server app](https://github.com/nextcloud/notes/issues/) nor the [Nextcloud Android app](https://github.com/nextcloud/android/issues).

### Describe the bug

Clicking on the favorite icon of a note in grid mode neither adds nor removes a note from the favorite list. I checked this via the file view on the Nextcloud server / app.

I debugged the issue in the code and nailed it down to the following line: 
https://github.com/nextcloud/notes-android/blob/8e225c79ff0f586260b62aa07dff1d2865d95b96/app/src/main/java/it/niedermann/owncloud/notes/main/MainActivity.java#L782

If this line is called, the method `toggleFavoriteAndSync` in class `NotesRepository` is called again which leads at the end to a neutral operation. 
https://github.com/nextcloud/notes-android/blob/8e225c79ff0f586260b62aa07dff1d2865d95b96/app/src/main/java/it/niedermann/owncloud/notes/persistence/NotesRepository.java#L524 

I tried to understand the reason why adding an observer leads to such a behavior but I did not succeed. Maybe there is somebody who can explain it to me. However, I also do not understand why this line is necessary at all because there is no place in the code where an observer is added to the live data. Hence, it is not necessary to remove one. If I remove the line, the app behaves as expected.

### Notes Android version

4.3.1

### Notes server version

4.11.0

### Nextcloud Android version

3.30.6

### Nextcloud version

30.0.4

### Device

Google Pixel 4a

### Android Version

13

### App Store

- [ ] Google Play Store
- [x] F-Droid
- [ ] Huawei App Gallery

**Comment by nowheretobefound** [comment]
I have the same issue

**Comment by chrgernoe** [comment]
For me the issue is solved in version 4.4.0. I guess via #2723.

**Comment by benjuade** [comment]
> For me the issue is solved in version 4.4.0. I guess via #2723.

Yes, it works but now introduced a new bug: the note gets emptied -.- (now trying to recover its content)

**Comment by chrgernoe** [comment]
Not on my side.

**Comment by chrgernoe** [comment]
So, from my side this issue can be closed.
